routes/index.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const router = express.Router();
+const db = require('../config/db');
+
+router.get('/', function (req, res) {
+    const sql = 'SELECT * from specializations';
+    db.query(sql, function (err, result) {
+        if (err) throw err;
+        res.render('index', { specializations: result });
+    });
+});
+
+router.post('/getticket', function (req, res) {
+    sql = 'INSERT INTO `clients` (' +
+        '`client_id`, `specialist_id`, `name`, `wasServed`, `ticketCreated`) ' +
+        'VALUES (NULL, ' + req.body.specialistSelect + ', \' ' + req.body.name + ' \', 0,  current_timestamp());';
+    console.log(sql);
+    db.query(sql, function (err, result) {
+        if (err) throw err;
+        res.redirect('/waitingList/'+result.insertId);
+    });
+});
+
+router.get('/waitingList', async function (req, res) {
+    var clients = [];
+    const sql = 'SELECT * from `clients` WHERE `wasServed`=0 LIMIT 20';
+    db.query(sql, async function (err, result) {
+        if (err) throw err;
+        for (var i = 0; i < result.length; i++) {
+            var client = {
+                name: result[i].name,
+                ticketCreated: result[i].ticketCreated,
+                number: result[i].client_id
+            };
+            var sql1 = 'SELECT name FROM `specialists` WHERE `specialist_id`=' + result[i].specialist_id;
+            db.query(sql1, function (err, specialists) {
+                if (err) console.log(err);
+                client.specialist = specialists[0].name;
+                clients.push(client);
+            });
+            await sleep(10);
```

Comment:
This `sleep` should not be required here. All queries should complete, then `res.render()` should happen.
Because `db.query()` is not blocking execution of further code, you need to consider a callback to be used instead of `sleep()` and `setTimeout()`.

Another improvement to make it more simple - use SQL queries to JOIN tables into one query and result, so you wouldn't have to call DB in a `for ()` cycle. Example `SELECT * FROM table_a INNER JOIN table_b on table_a.id == table_b.other_id` (haven't checked the actual syntax, but idea is somewhat similar.

Reply:
If you'd like to keep multiple queries and `for` cycle, you'd need to consider using `Promise` and `Promise.all([/*all promises from for cycle in this array*/])`